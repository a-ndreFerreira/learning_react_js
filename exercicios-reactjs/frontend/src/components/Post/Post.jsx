import './Post.css'

import PostCard from '../PostCard/PostCard';

const Post = ({ posts }) => {


    return (
        <div className='posts'>
            {
                posts?.map((item) => {
                    const { body, title, id, cover } = item;

                    return (
                        <PostCard
                            key={id}
                            id={id}
                            cover={cover}
                            title={title}
                            body={body}
                        />
                    )
                })
            }
        </div>
    )
}

export default Post