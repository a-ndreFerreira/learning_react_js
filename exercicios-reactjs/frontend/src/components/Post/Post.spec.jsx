import { render, screen } from '@testing-library/react'
import { describe, expect } from 'vitest'

import Post from './Post'

const posts = {
    posts: [
        {
            title: 'title 1',
            body: 'body 1',
            id: 1,
            cover: 'img/img1.png'
        },
        {
            title: 'title 2',
            body: 'body 2',
            id: 2,
            cover: 'img/img2.png'
        },
        {
            title: 'title 3',
            body: 'body 3',
            id: 3,
            cover: 'img/img3.png'
        },
    ]
};

describe('<Post />', () => {
    it('should be render all the props an component in screen', () => {
        // const { debug } = render(<Post {...posts} />);

        // debug();

        render(<Post {...posts} />);

        expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);

        expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);

        expect(screen.getAllByText(/body/i)).toHaveLength(3);

        expect(screen.getByRole('img', { name: 'title 3' })).toHaveAttribute('src', 'img/img3.png');

    })

    it('should match snapshot', () => {
        const { container } = render(<Post {...posts} />);

        expect(container.firstChildren).toMatchSnapshot();
    })

})