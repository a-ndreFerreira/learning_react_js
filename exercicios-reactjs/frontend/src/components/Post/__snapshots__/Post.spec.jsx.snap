// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Post /> > should match snapshot 1`] = `undefined`;
